### ISSUE TITLE
what do we want from the website?

### ISSUE THREAD

**Issue by bcoe** [yargs/yargs.github.io#1]
some things I'd love:

* a landing page that has a quick-start guide, and shows off some big projects using yargs.
* the docs (preferably we'd render the README.md directly from a yargs dependency, to avoid @nexdrew's fear of documentation getting out of sync with yargs itself).
* community page where people can show off neat modules they've made that work with yargs.

**Comment by bcoe** [comment]
## Prior Art

I dug into what some of the other Node.js libraries are using for their website:

[expressjs](https://github.com/strongloop/expressjs.com): 

  * uses jekyll. 
  * Has a really clean design, I think there's a lot we can learn from.

[yeoman](https://github.com/yeoman/yeoman.github.io) 

  * uses jekyll with a Grunt build process. 
  * Beautiful website with lots of illustrations. With some of the [amazing designs]
(https://github.com/bcoe/yargs/issues/324) @chrisdickinson has been putting together for yargs, we might be able to emulate some of yeoman's artistic acumen.
  * I like how it lists the contributors on the front-page.

[gulpjs](https://github.com/gulpjs/gulpjs.github.io):
  * uses `gulp-gh-pages`.
  * really simple design, basically just a big button that says try gulp.

[browserify](https://github.com/substack/browserify-website):
  * uses static HTML ... I like how simple this is.
  * @substack puts a lot of great thought into his information architecture, I think we can learn a lot from this site.

Some of my initial take aways:

* we should pin down exactly what information we want on the website (docs? tutorials? showcase?).
* given that a lot of our documentation is already in Markdown, it seems like jekyll would be a reasonable choice for rendering content.
* I have a [bare-bones Node.js static-site generator](https://github.com/polyglotweekly/articleify) I've put some work into, if we're feeling adventurous we could try to build this out further.

Hope this gets people thinking, curious what platform other folks would be comfortable with.

**Comment by maxrimue** [comment]
I too think that Jekyll seems like the best choice here, since it's already built into GitHub pages and allows us to write both Markdown and HTML files for a static site. For example, we could have an `index.html` giving a brief introduction to yargs and real quick installation/getting started instructions, and on that page we could link to a Readme markdown file, and another HTML or Markdown file for projects using yargs (which can be updated by pull requests).

Basically, the `Readme.md` would have the same content like the `Readme.md` from the yargs repo, we'd only have to wrap the pure Readme Markdown code in some HTML/Markdown code for adding for example a header, which makes sense on a website, but not in the Readme in the yargs repo

**Comment by lrlna** [comment]
A break down of sorts: 

### Home:
- [ ] [animated] pirate
- [ ] overview of yargs: why we cool etc.
- [ ] REPL, as suggested by @nexdrew ( could also be under [docs](### Docs)? )
- [ ] Basic example: e.g. part on [options](https://github.com/bcoe/yargs#tell-users-how-to-use-yer-options-and-make-demands)

### Community Projects:
This could be handled in:
- repl form for folks to use projects (+ npm/github links)
- a grid of different projects (+ npm/github links) -- A block on the grid could be as simple as `npm install [package]` + `authors` ? or something fancier with images and animations and cool thangs
- actual usage-of-yargs examples from these projects (might be an overkill tbh)

### Docs
- [ ] Full scale docs of all the yargs things

**Comment by lrlna** [comment]
Ugh, sorry, ^ that's p much me repeating what @bcoe said in the first place, but with checkboxes :disappointed:

**Comment by nexdrew** [comment]
@lrlna Don't worry about it - the more organization via bullet points, checkboxes, and tasks, the better! :+1: 

I know we haven't made any decisions yet, but if we're gonna host this thing via GitHub Pages then we'll need to rename this repository to `yargs.github.io`.

Also, if we want to keep using the **free** DNS name `yargs.js.org` (thanks to the [JS.ORG team](https://github.com/js-org)), then we'll need to remember to [update their repo](https://github.com/js-org/dns.js.org/blob/master/cnames_active.js) to point to the new GH Pages URL too.

Just saying. :grin:

**Comment by nexdrew** [comment]
Here's that last comment in checkbox form :smile_cat: (assuming use of GH Pages):

- [ ] Rename this repo to `yargs.github.io` (and add content to it)
- [ ] DNS 1/3: Remove CNAME file from current [gh-pages branch](https://github.com/bcoe/yargs/tree/gh-pages)
- [ ] DNS 2/3: Add CNAME file to this repo
- [ ] DNS 3/3: Update GH Pages URL in [dns.js.org repo](https://github.com/js-org/dns.js.org)

**Comment by lrlna** [comment]
:+1: to hosting this with github pages. Seems most convenient :100:

**Comment by bcoe** [comment]
@lrlna this is great, :white_check_mark: all the things \o/

**Comment by lrlna** [comment]
@nexdrew I ought to PR on [dns.js.org](https://github.com/js-org/dns.js.org/blob/master/cnames_active.js) to change it to yargs.github.io, right?

**Comment by nexdrew** [comment]
@lrlna Yes, but the GH Pages site must be up first.

**Comment by nexdrew** [comment]
@lrlna Also, the CNAME should be changed to `yargs.js.org` first :+1:
